examples/parser-tests.dx

```diff
@@ -38,3 +38,41 @@ f = \x. x + 10.
 
 :p f (-1.0)
 > 9.0
+
+'Lambdas can have specific arrow annotations.
+
+lam1 = \n ?-> \x. fromOrdinal n x
+:t lam1
+> ((n:Type) ?-> Int32 -> n)
+
+lam2 = \x --o 2.0 * x
+:t lam2
+> (Float32 --o Float32)
+
+lam3 = \d:(Eq Int) ?=> ()
+:t lam3
+> ((Eq Int32) ?=> Unit)
+
+lam4 = \n m ?-> (0@n, 0@m)
```

Comment:
Hmm is this the same convention that we use in `def`s? I thought that there you have to annotate each argument with its arrow, and it's not like the choice propagates back. Or is it that one can never mix multiple arrow types in a single lambda, and you have to declare a nested one if you want a different arrow type?

Reply:
Yeah, this is a different convention than `def`s; I figured that just declaring a nested lambda if you want to change arrows isn't that much of an inconvenience.

An alternative would have been to make lambdas act like defs and always require a closing `.`, so that you would write something like `lam4 = \n?-> m?->. (0@n, 0@m)`. But that seemed harder to read, at least in my opinion.